### PR TITLE
Prompt user to install Neo Express when needed; explicitly request JSON output from `neo-express show` commands

### PR DESCRIPTION
**Pull request by djnicholson**
Fixes https://github.com/neo-project/neo-visual-tracker/issues/28
Fixes https://github.com/neo-project/neo-visual-tracker/issues/29

### CODE REVIEW

File: src/neoExpressHelper.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        return await new Promise((resolve) => {
+            childProcess.exec(command, (error, stdout, stderr) => {
+                if (error) {
+                    console.error('Error showing account', command, error, stderr);
+                    resolve(JsonResult.Error());
+                } else if (stderr) {
+                    console.error('Error showing account', command, error, stderr);
+                    resolve(JsonResult.Error());
+                } else {
+                    resolve(JsonResult.Success(stdout));
```
Comment (devhawk) [inline src/neoExpressHelper.ts]: does this json content get parsed elsewhere?

File: src/neoExpressHelper.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    }
+
+    public static Success(outputJson: string) {
+        return new JsonResult(false, JSON.parse(outputJson));
+    }
+}
+
+export class NeoExpressHelper {
+
+    public static async claimGas(
```
Comment (devhawk) [inline src/neoExpressHelper.ts]: Thinking we should add --json option (even if it does nothing at first) to express claim and transfer commands. THoughts? Opened https://github.com/neo-project/neo-express/issues/26 to track

File: src/installRequiredPanel.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+const JavascriptHrefPlaceholder : string = '[JAVASCRIPT_HREF]';
+const CssHrefPlaceholder : string = '[CSS_HREF]';
+
+export class InstallRequiredPanel {
```
Comment (devhawk) [inline src/installRequiredPanel.ts]: Do we need a custom UI for this? What I've typically seen in VSCode is use an information message. here's an [example from Powershell extension](https://github.com/PowerShell/vscode-powershell/blob/dba7d83e91ab2b0d1e359f8f22237450437c1f17/src/features/UpdatePowerShell.ts#L79). I especially like how it provides the user the option to update now, not update now but get reminded in future, and not to get bugged about it ever again.

File: src/neoExpressHelper.ts (hunk 1 above)
Comment (djnicholson) [reply src/neoExpressHelper.ts]: it's parsed in the constructor for JsonResult

File: src/neoExpressHelper.ts (hunk 2 above)
Comment (djnicholson) [reply src/neoExpressHelper.ts]: Yes, I can't start passing it in as an argument until after you add support for it though (or the command will error out with "Unrecognized option '--json'")

File: src/installRequiredPanel.ts (hunk 3 above)
Comment (djnicholson) [reply src/installRequiredPanel.ts]: good call. probably not right to have a "don't remind me ever again" option in this case though (as the UI would just seem broken for no reason after selecting that, and the user may forget why)

File: src/neoExpressHelper.ts (hunk 2 above)
Comment (devhawk) [reply src/neoExpressHelper.ts]: Given #33, I think we should *not* do this

File: src/installRequiredPanel.ts (hunk 3 above)
Comment (devhawk) [reply src/installRequiredPanel.ts]: Maybe we prompt to install when the user attempts to start a neo express instance via right click or the play button. Then the options would be "install" or "cancel" with a clear user action taken to trigger the prompt again

Comment (djnicholson) [reply src/installRequiredPanel.ts]: That's what this currently does, except that the options are "Cancel" (I don't get to choose the wording on that button) and "Show installation instructions" (do you want that to say "Install" instead?)

I also have the same behavior for the other dialogs that currently invoke neo-express, but that's the right thing to do for now and can be removed as the neo-express dependency is removed from each of those dialogs.

File: src/neoExpressHelper.ts (hunk 2 above)
Comment (djnicholson) [reply src/neoExpressHelper.ts]: agree

File: src/installRequiredPanel.ts (hunk 3 above)
Comment (devhawk) [reply src/installRequiredPanel.ts]: Could we have Cancel, "Install" (which simply invokes `dotnet tool install Neo.Express -g`) and "More Info" (which does what showInstructions does now)?

File: src/extension.ts [hunk 4]
```diff
@@ -20,6 +21,21 @@ export function activate(context: vscode.ExtensionContext) {
...
+			const dialogResponse = await vscode.window.showInformationMessage(
+				'Neo Express Required\n\nNeo Express was not detected on your machine. Neo Express must be installed in order to use this functionality.\n',
+				{ modal: true },
+				showInstructions);
+			if (dialogResponse === showInstructions) {
+				vscode.env.openExternal(vscode.Uri.parse('https://github.com/neo-project/neo-express#Installation'));
+			}
+		}
+	};
+
```
Comment (devhawk) [inline src/extension.ts]: I like this much better. Should we raise a notification when the install is done? Also, do we continue with the original task after install is complete?

Comment (djnicholson) [reply src/extension.ts]: we didn't... i just changed it so that we do...

**Pull request merged**